lib/dash/dash_parser.js

```diff
@@ -868,15 +857,26 @@ shaka.dash.DashParser = class {
   }
 
   /**
+   * Gets the streams from the given sets or returns an empty array if disabled
+   * or no streams are found.
+   * @param {boolean} disabled
    * @param {!Array.<!shaka.dash.DashParser.AdaptationInfo>} adaptationSets
-   * @param {string} type
-   * @return {!Array.<!shaka.dash.DashParser.AdaptationInfo>}
-   * @private
-   */
-  getSetsOfType_(adaptationSets, type) {
-    return adaptationSets.filter((as) => {
-      return as.contentType == type;
-    });
+   * @param {string} contentType
+   @private
+  */
+  getStreamsFromSets_(disabled, adaptationSets, contentType) {
+    if (disabled || !adaptationSets.length) {
+      return [];
+    }
+
+    return adaptationSets.reduce((acc, currSet) => {
```

Comment:
You could also use `shaka.util.Functional.collapseArrays` for this.
Something like...

```js
return adaptationSets.map((set) => {
  return set.contentType == contentType ? set.streams : [];
}).reduce(shaka.util.Functional.collapseArrays, []);
```

If you think that doing it this way is better, though, instead I'll ask that you rename the variables on the method to `all` and `part`, as that is what we usually call them when reducing arrays.

Reply:
changed to `(all, part)`